grpc/doc/swdesign/README.md

```diff
@@ -45,13 +51,190 @@ Considered alternatives:
 * implementing the gRPC protocol completely inside the middleware
 * using an http library and a protobuf converter
 
+#### Support PEM file format for X509 certificates
+`swdd~grpc-supports-pem-file-format-for-X509-certificates~1`
+
+Status: approved
+
+When a PEM file is provided as certificate, the gRPC middleware shall:
+* parse the given PEM file into x509 certificate buffer format.
+* use the parsed x509 certificate to setup the TLS configuration.
+
+Rationale:
+
+The advantage of using a PEM file is due to its text-based, human-readable format, making it more versatile, as it can contain certificates, private keys, public keys and even certificate chain, compared to DER.
```

Comment:
What about writing this as design decision?

Reply:
It actually already is a design decision. I'll extract the alternatives at the correct place to make this more prominent.

Reply:
The reqs here are moved and converted to normal requirements. 
A design decision on this level is a bit lost.